### PR TITLE
chore: docs updates and use some defaults

### PR DESCRIPTION
**Pull request by incertum**
- Docs updates
- Rely on default cri socket search logic
- Trust the default driver-loader as we the CNCF testbed is ubuntu 22.04 for which we definitely always have pre-built drivers. For the records: I cannot confirm that driver-loader would work building the driver as backup.

### CODE REVIEW

File: kustomize/falco-driver/ebpf/daemonset.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ spec:
           touch /tmp/falco/events.jsonl &&
           logrotate -f /etc/falco/logrotate.conf &&
           export FALCO_HOSTNAME="${FALCO_HOSTNAME}_${CONFIG_VERSION}" &&
-          /usr/bin/falco --cri /run/containerd/containerd.sock -pk --disable-cri-async
+          /usr/bin/falco -pk --disable-cri-async
```
Comment (incertum) [inline kustomize/falco-driver/ebpf/daemonset.yaml:63]: Better to rely on the defaults.

File: kustomize/falco-driver/kmod/daemonset.yaml [hunk 2]
```diff
@@ -84,23 +84,8 @@ spec:
...
         image: docker.io/falcosecurity/falco-driver-loader:master
         imagePullPolicy: IfNotPresent
-        # args: 
-        # - kmod
-        # todo we are still investigating driver-loader issues in general and better ways to support testing w/ minikube
-        # consider below override experimental for now to support minikube testing as well
-        command: ["/bin/bash", "-c"]
-        args:
-        - |
-          if [ "$(ls -A ${HOST_ROOT}/usr/src/kernels/$(uname -r))" ]; then
```
Comment (incertum) [inline kustomize/falco-driver/kmod/daemonset.yaml:94]: The strangest thing happened again. yesterday on Jan 10 it all worked and today pulling the latest driver-loader master it again just didn't work somehow with the mounts (everything the same). I had this before (changing behavior before the XMas break compared to yesterday), perhaps we have a Schroedingers cat here, I truly don't know.

Just for the records here the output I still had in my terminal from yesterday:

```

root@falco-driver-kmod-4f6dp:/# if [ "$(ls -A ${HOST_ROOT}/usr/src/kernels/$(uname -r))" ]; then
              echo "RPM based distro"
              mkdir -p /usr/src/kernels/ && cp -r ${HOST_ROOT}/usr/src/kernels/$(uname -r) /usr/src/kernels/$(uname -r) &&
              /usr/bin/falcoctl driver config --type kmod && /usr/bin/falcoctl driver install --compile=true --download=false
          else
              echo "Not RPM based distro"
              cp -r $(ls -d ${HOST_ROOT}/usr/src/* | head -n 1) /usr/src/ &&
              cp -r $(ls -d ${HOST_ROOT}/usr/src/linux-headers-$(uname -r) | head -n 1) /usr/src/linux-headers-$(uname -r) &&
              /usr/bin/falcoctl driver config --type kmod && /usr/bin/falcoctl driver install --compile=true --download=false
          fi
RPM based distro
2024-01-10 06:39:00 INFO  Running falcoctl driver config
                      ├ name: falco
                      ├ version: 7.0.0+driver
                      ├ type: kmod
                      ├ host-root: /host
                      └ repos: https://download.falco.org/driver
2024-01-10 06:39:00 INFO  Running falcoctl driver install
                      ├ driver version: 7.0.0+driver
                      ├ driver type: kmod
                      ├ driver name: falco
                      ├ compile: true
                      ├ download: false
                      ├ arch: x86_64
                      ├ kernel release: 6.2.15-100.fc36.x86_64
                      └ kernel version: #1 SMP PREEMPT_DYNAMIC Thu May 11 16:51:53 UTC 2023
2024-01-10 06:39:00 INFO  Found distro target: ubuntu-generic
2024-01-10 06:39:00 INFO  Check if kernel module is still loaded.                                                                                                                                      
2024-01-10 06:39:00 INFO  Kernel module is still loaded. 
2024-01-10 06:39:00 INFO  Trying to unload it with 'rmmod'. 
2024-01-10 06:39:00 INFO  OK! Unloading module succeeded. 
2024-01-10 06:39:00 INFO  Check all versions of kernel module in dkms. 
2024-01-10 06:39:00 INFO  OK! There are no module versions in dkms. 
2024-01-10 06:39:00 INFO  Trying to dkms install module. gcc: /usr/bin/gcc                                                                                                                             
2024-01-10 06:39:14 INFO  Module installed in dkms. file: /var/lib/dkms/falco/7.0.0+driver/6.2.15-100.fc36.x86_64/x86_64/module/falco.ko.xz
2024-01-10 06:39:14 INFO  Copying built driver to its destination.
                      ├ src: /var/lib/dkms/falco/7.0.0+driver/6.2.15-100.fc36.x86_64/x86_64/module/falco.ko.xz
                      └ dst: /root/.falco/7.0.0+driver/x86_64/falco_ubuntu-generic_6.2.15-100.fc36.x86_64_1.ko
2024-01-10 06:39:14 INFO  Driver built. path: /root/.falco/7.0.0+driver/x86_64/falco_ubuntu-generic_6.2.15-100.fc36.x86_64_1.ko
2024-01-10 06:39:14 INFO  Success: module found and loaded in dkms.
                      └ driver: /root/.falco/7.0.0+driver/x86_64/falco_ubuntu-generic_6.2.15-100.fc36.x86_64_1.ko
root@falco-driver-kmod-4f6dp:/# lsmod | grep falco
falco                 966656  0
```

Also I actually checked on a prod server and since the driver build script is looking for `/lib/modules` I think it shouldn't work as it's mounted to `/host/lib/modules`. We will investigate that once we have the new helm chart. 

In summary, regardless of minikube and mounts I am not sure building the driver works easily. Given on the CNCF cluster we have ubuntu 22.04 we are good to guarantee we always have a pre-built driver.

@leogr officially giving up on confirming driver-loader can work building the driver in cases we have no pre-built driver. Hope that is acceptable.

Comment (leogr) [reply kustomize/falco-driver/kmod/daemonset.yaml:94]: > Hope that is acceptable.

It is for me.

**Review by leogr — APPROVED** [review]
